std/algorithm/iteration.d

```diff
@@ -888,66 +889,70 @@ template each(alias pred = "a")
         (!isForwardRange!R || isDynamicArray!R) &&
         (isForeachUnaryIterable!R || isForeachBinaryIterable!R);
 
+public:
     void each(Range)(Range r)
-    if (isRangeIterable!Range && !isForeachIterable!Range)
+    if (!isForeachIterable!Range && (
+        isRangeIterable!Range ||
+        __traits(compiles, typeof(r.front).length)))
```

Comment:
Did you mean the 1st and 4th overloads are grouped as one group, and the 2nd and 3rd as another group?  I'm looking at the code and the sig constraints of the 1st and 3rd, taken together, don't match what you have here.